src/qibolab/instruments/qblox/cluster.py

```diff
@@ -0,0 +1,150 @@
+""" Qblox instruments driver.
+
+Supports the following Instruments:
+    Cluster
+    Cluster QRM-RF
+    Cluster QCM-RF
+    Cluster QCM
+Compatible with qblox-instruments driver 0.9.0 (28/2/2023).
+It supports:
+    - multiplexed readout of up to 6 qubits
+    - hardware modulation, demodulation, and classification
+    - software modulation, with support for arbitrary pulses
+    - software demodulation
+    - binned acquisition
+    - real-time sweepers of
+        - pulse frequency (requires hardware modulation)
+        - pulse relative phase (requires hardware modulation)
+        - pulse amplitude
+        - pulse start
+        - pulse duration
+        - port gain
+        - port offset
+    - multiple readouts for the same qubit (sequence unrolling)
+    - max iq pulse length 8_192ns
+    - waveforms cache, uses additional free sequencers if the memory of one sequencer (16384) is exhausted
+    - instrument parameters cache
+    - safe disconnection of offsets on termination
+
+The operation of multiple clusters simultaneously is not supported yet.
+https://qblox-qblox-instruments.readthedocs-hosted.com/en/master/
+"""
+
+from dataclasses import dataclass
+from enum import Enum
+
+from qblox_instruments.qcodes_drivers.cluster import Cluster as QbloxCluster
+
+from qibolab.instruments.abstract import Instrument, InstrumentException
+
+
+class ReferenceClockSource(Enum):
+    INTERNAL = "internal"
+    EXTERNAL = "external"
+
+
+@dataclass
+class Cluster_Settings:
+    """A class to store the settings of the Cluster instrument.
+
+    Attributes:
+        reference_clock_source (:class:`qibolab.instruments.qblox.cluster.ReferenceClockSource`):
+            (ReferenceClockSource.INTERNAL, ReferenceClockSource.EXTERNAL) Instructs the cluster to use the internal
+            clock source or an external source.
+    """
+
+    reference_clock_source: ReferenceClockSource = ReferenceClockSource.INTERNAL
```

Comment:
Since this dataclass has a single attribute we could even drop it (for simplicity) and move the `reference_clock_source` to the `Cluster`, but I understand if you prefer to keep it for symmetry with the other qblox instrument objects.

Reply:
Yes, it was introduced for symmetry. Let's keep it as it is.